My Recipes/scr/Controllers/AddRecipeViewController.swift

```diff
@@ -0,0 +1,316 @@
+        case .instuction:
+            cell.textLabel?.text = Strings.instruction
+            cell.accessoryType = .disclosureIndicator
+        }
+    }
+    
+    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        let isSaved = recipeId != nil
+        
+        let index = CellIndex(rawValue: indexPath.row) ?? .recipeName
+       
+        switch index {
+        case .recipeName, .duration, .servers, .complexity:
+            return UITableView.automaticDimension
+        default:
+            return isSaved ? 44 : 0
+        }
+    }
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        
+        selectedIndex = CellIndex(rawValue: indexPath.row)
+        
+        switch selectedIndex ?? .recipeName {
+        case .recipeName:
+            presentInputAlertView(title: Messages.enterRecipeName) { value in
+                self.recipeName = value
+                self.tableView.reloadData()
+            }
+            break
+        case .recipePhoto:
+            openImageSelection()
+            break
+        case .duration:
+            presentInputAlertView(title: Messages.enterDuration) { value in
+                self.duration = value
+                self.tableView.reloadData()
+            }
+            break
+        case .complexity:
+            break
+        case .servers:
+            break
+        case .ingredients:
+            let intance = RecipeIngredientsVC.instance
+            intance.recipeId = recipeId
+            navigationController?.pushViewController(intance, animated: true)
+            break
+        case .instuction:
+            let intance = RecipeInstructionVC.instance
+            intance.recipeId = recipeId
+            navigationController?.pushViewController(intance, animated: true)
+            break
+        }
+        
+    }
+}
+
+// MARK: - Selectors
+extension AddRecipeViewController{
+    @IBAction func servesStepperValueChanged(){
+        let people = Int(servesStepper.value)
+        lblNoOfServes.text = "\(people) \(people == 1 ? "People" : "Peoples")"
+    }
+    
+    @objc func segmentValueChanged(_ sender:UISegmentedControl){
+        complexityIndex = sender.selectedSegmentIndex
+    }
+    
+    @objc func saveTapped(){
+        let params: [String:Any] = [WSParams.name: recipeName ?? "",
+                                    WSParams.preparationTime: duration ?? "",
+                                    WSParams.serves: Int(serves),
+                                    WSParams.complexity: complexityArray[complexityIndex]]
+        indicator.startAnimation()
+        recipeProvider.request(.add(params: params)) { result in
+            self.indicator.stopAnimation()
+            switch result{
+            case let .success(response):
+                do{
+                    struct Response: Codable {
```

Comment:
Codable struct can be defined publically so it can be used anywhere in the project.